app/main.py

```diff
@@ -1,8 +1,29 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
 
+    def __init__(self, name: str, age: int) -> None:
+        self.name = name
+        self.age = age
+        Person.people[self.name] = self
 
-def create_person_list(people: list) -> list:
-    # write your code here
-    pass
+
+def create_person_list(peoples: list) -> list:
+    person_list = []
+    for pers in peoples:
+        person = Person(pers["name"], pers["age"])
+        person_list.append(person)
+    for pers in peoples:
+        person = Person.people[pers["name"]]
+        if "wife" in pers and pers["wife"] is not None:
+            wife = Person.people.get(pers["wife"])
+            if wife:
+                setattr(person, "wife", wife)
+            else:
+                setattr(person, "wife", None)
+        if "husband" in pers and pers["husband"] is not None:
+            husband = Person.people.get(pers["husband"])
+            if husband:
+                setattr(person, "husband", husband)
+            else:
+                setattr(person, "husband", None)
```

Comment:
The current logic for setting 'wife' and 'husband' attributes assumes that these names will always be present in the `Person.people` dictionary. If a name is not found, `None` will be set, which might not be the desired behavior. Adding error handling for cases where the spouse is not found will make your code more robust and reliable.